pkgs/development/python-modules/torch/default.nix

```diff
@@ -52,15 +52,15 @@
 
 let
   inherit (lib) lists strings trivial;
-  inherit (cudaPackages) cudatoolkit cudaFlags cudnn nccl;
+  inherit (cudaPackages) cudaFlags cudnn nccl;
 in
 
 assert cudaSupport -> stdenv.isLinux;
 assert cudaSupport -> (cudaPackages.cudaMajorVersion == "11");
 
 # confirm that cudatoolkits are sync'd across dependencies
-assert !(MPISupport && cudaSupport) || mpi.cudatoolkit == cudatoolkit;
-assert !cudaSupport || magma.cudaPackages.cudatoolkit == cudatoolkit;
+assert !(MPISupport && cudaSupport) || mpi.cudatoolkit == cudaPackages.cudatoolkit;
+assert !cudaSupport || magma.cudaPackages.cudaVersion == cudaPackages.cudaVersion;
```

Comment:
This is a different behaviour though, it's a weaker constraint compared to `a.cudatoolkit == b.cudatoolkit`: the original check was to ensure that overlays have been applied consistently (one could apply overrides to `a` and not to `b`, which would result in having two cudatoolkits and the build closure). We might want to consider: 1) replacing the check with `a.cudaPackages == b.cudaPackages` (I'm not sure what is the complexity of such test), 2) adding a comment, explaining why the test is there, 3) moving the test to the `meta.broken` so the end user may override it, 4) perhaps even dropping the test altogether?

Note that this test is ad hoc and doesn't guarantee that conflicting cuda components don't end up in the closure some other way. If later we want to ensure that doesn't happen, we might try handle this at the setup hook level

Reply:
Fixed in latest.